### PR TITLE
Cherrypick - Package .so files in AAR File (#933)

### PR DESCRIPTION
**Pull request by rasaha91**
#### Please select one of the following
- [ ] I am removing an existing difference between facebook/react-native and microsoft/react-native-macos :thumbsup:
- [ ] I am cherry-picking a change from Facebook's react-native into microsoft/react-native-macos :thumbsup:
- [ ] I am making a fix / change for the macOS implementation of react-native
- [x] I am making a change required for Microsoft usage of react-native

## Summary

This change was made in 0.64 but never cherry picked to main. As a result, it was not made available to 0.66 which is leading to crashes for consumers of 0.66.

## Changelog

[Android] [Fixed] - Cherry-pick change to package .so files in AAR

### CODE REVIEW

File: .ado/publish.yml
Comment (mganandraj) [inline .ado/publish.yml:213]: This is used for nuget package creation to be consumed in office .. So this should excluded the libraries .. the gradle invocation in npmOfficePack.js is used for npm package publish.

**Review by mganandraj — APPROVED** [review]
Approving assuming you will rollback the changes in publish.yml

File: .ado/publish.yml
Comment (rasaha91) [reply .ado/publish.yml:213]: As discussed offline, this is just a cherry pick of @msgharpu 's change

Comment (rasaha91) [reply .ado/publish.yml:213]: Will wait on @mganandraj to confirm before merging.

**Pull request closed without merging**
